src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
Hmm, maybe I am misunderstanding, but shouldn't it always end up with an even split since it's a hyperplane and the similarity metric won't define what side of the hyperplane a point lies? I think what you are worried about is that it could unfairly split similar points because they could potentially seem like they are far from each when you project onto the randomly chosen plane. ie: if the angle between 2 points was small but the magnitude was large. In that case, as long as you normalize ahead of time it should be a more fair split since then all points will lie on the unit sphere. Might need to think about this some more. 

Good point about the simplicity and avoiding any determinism in the tree splits. Not sure on the theoretical implications of smarter splits, but seems like there will still be plenty of randomness from the initial hyperplane choice as well as the samples.